File: app/src/main/java.backup/com/lumiyaviewer/lumiya/slproto/messages/AssetUploadRequest.java
package com.lumiyaviewer.lumiya.slproto.messages;

import com.lumiyaviewer.lumiya.slproto.SLMessage;
import java.nio.ByteBuffer;
import java.util.UUID;

public class AssetUploadRequest extends SLMessage {
    public AssetBlock AssetBlock_Field = new AssetBlock();

    public static class AssetBlock {
        public byte[] AssetData;
        public boolean StoreLocal;
        public boolean Tempfile;
        public UUID TransactionID;
        public int Type;
    }

    public AssetUploadRequest() {
        this.zeroCoded = false;
    }

    public int CalcPayloadSize() {
        return this.AssetBlock_Field.AssetData.length + 21 + 4;
    }

    public void Handle(SLMessageHandler sLMessageHandler) {
        sLMessageHandler.HandleAssetUploadRequest(this);
    }

    public void PackPayload(ByteBuffer byteBuffer) {
        byteBuffer.putShort(-1);
        byteBuffer.put((byte) 1);
        byteBuffer.put((byte) 77);
        packUUID(byteBuffer, this.AssetBlock_Field.TransactionID);
        packByte(byteBuffer, (byte) this.AssetBlock_Field.Type);
        packBoolean(byteBuffer, this.AssetBlock_Field.Tempfile);
        packBoolean(byteBuffer, this.AssetBlock_Field.StoreLocal);
        packVariable(byteBuffer, this.AssetBlock_Field.AssetData, 2);
    }

    public void UnpackPayload(ByteBuffer byteBuffer) {
        this.AssetBlock_Field.TransactionID = unpackUUID(byteBuffer);
        this.AssetBlock_Field.Type = unpackByte(byteBuffer);
        this.AssetBlock_Field.Tempfile = unpackBoolean(byteBuffer);
        this.AssetBlock_Field.StoreLocal = unpackBoolean(byteBuffer);
        this.AssetBlock_Field.AssetData = unpackVariable(byteBuffer, 2);
    }
}

File: lumiya/src/main/java/com/lumiyaviewer/lumiya/slproto/messages/AssetUploadRequest.kt
package com.lumiyaviewer.lumiya.slproto.messages

import com.lumiyaviewer.lumiya.slproto.SLMessage
import com.lumiyaviewer.lumiya.slproto.handler.SLMessageHandler
import java.nio.ByteBuffer
import java.util.UUID

class AssetUploadRequest : SLMessage {
    var AssetBlock_Field: AssetBlock = AssetBlock()

    class AssetBlock {
        var AssetData: ByteArray = ByteArray(0)
        var StoreLocal: Boolean = false
        var Tempfile: Boolean = false
        var TransactionID: UUID? = null
        var Type: Int = 0
    }

    constructor() {
        this.zeroCoded = false
    }

    override fun CalcPayloadSize(): Int {
        return AssetBlock_Field.AssetData.size + 21 + 4
    }

    override fun Handle(sLMessageHandler: SLMessageHandler) {
        sLMessageHandler.HandleAssetUploadRequest(this)
    }

    override fun PackPayload(byteBuffer: ByteBuffer) {
        byteBuffer.putShort(-1)
        byteBuffer.put(1.toByte())
        byteBuffer.put(77.toByte())
        packUUID(byteBuffer, AssetBlock_Field.TransactionID!!)
        packByte(byteBuffer, AssetBlock_Field.Type.toByte())
        packBoolean(byteBuffer, AssetBlock_Field.Tempfile)
        packBoolean(byteBuffer, AssetBlock_Field.StoreLocal)
        packVariable(byteBuffer, AssetBlock_Field.AssetData, 2)
    }

    override fun UnpackPayload(byteBuffer: ByteBuffer) {
        AssetBlock_Field.TransactionID = unpackUUID(byteBuffer)
        AssetBlock_Field.Type = unpackByte(byteBuffer).toInt()
        AssetBlock_Field.Tempfile = unpackBoolean(byteBuffer)
        AssetBlock_Field.StoreLocal = unpackBoolean(byteBuffer)
        AssetBlock_Field.AssetData = unpackVariable(byteBuffer, 2)
    }
}
